smarts/core/coordinates.py

```diff
@@ -235,3 +243,15 @@ def heading(self):
     def as_panda3d(self):
         """ Convert to panda3D (object bounds centre position, heading)"""
         return (self.position, self.heading.as_panda3d)
+
+    @classmethod
+    def from_ros(cls, ros_pose):
```

Comment:
I'm not sure what you mean here.  The parameter to this is of type [geometry_msgs/Pose](http://docs.ros.org/en/noetic/api/geometry_msgs/html/msg/Pose.html), a ROS message type.  (It would pollute things too much to add that as a type hint here.)
It's used in `ros_driver.py`.  Maybe you mean that I should move a version of this function there and not add it here?  If so, I think I can agree to that.  But I want to confirm first to make sure I understand your concern.

Reply:
What I mean specifically is that we purposefully avoided `from_<source>` calls because then we have a dependency on converting to their format within SMARTS rather than working with the coordinate frame.

Reply:
Ok, understood.  I just moved it into `ros_driver.py`.